### DISCUSSION TITLE
What iOS versions to support?

### DISCUSSION THREAD

**Discussion by louwers** [maplibre/maplibre-native#689] [General]
We have code to run MapLibre GL Native for iOS in iOS 9 simulator. This is probably cruft that we should get rid of.

Ideally we define which iOS versions are supported.

According to Apple, [4% of iPhones are on < iOS 14 and 1% of iPhones released in the last 4 years](https://developer.apple.com/support/app-store/).

I think it is reasonable to only support iOS 12 or above (following the recent move of WhatsApp).

-  [Here are the devices that still support iOS 14](https://en.wikipedia.org/wiki/IOS_14#Supported_devices)
- [iOS 13](https://en.wikipedia.org/wiki/IOS_13#Supported_devices)
- [iOS 12](https://en.wikipedia.org/wiki/IOS_12#Supported_devices) ([iPhone 5S](https://en.wikipedia.org/wiki/IPhone_5S), [6](https://en.wikipedia.org/wiki/IPhone_6), [6 Plus](https://en.wikipedia.org/wiki/IPhone_6), and the [iPod Touch (6th generation)](https://en.wikipedia.org/wiki/IPod_Touch_(6th_generation)), all have limited support.)
- [iOS 11](https://en.wikipedia.org/wiki/IOS_11#Supported_devices) (drops support for [iPhone 5](https://en.wikipedia.org/wiki/IPhone_5), [iPhone 5C](https://en.wikipedia.org/wiki/IPhone_5C), and [iPad 4](https://en.wikipedia.org/wiki/IPad_(4th_generation)), making it the first version of iOS to run exclusively on iOS devices with [64-bit processors](https://en.wikipedia.org/wiki/64-bit_computing).[](https://en.wikipedia.org/wiki/IOS_11#cite_note-143))

﻿﻿<img width="715" alt="grafik" src="https://user-images.githubusercontent.com/649392/212062933-7b4110fa-c35d-435e-a9b3-d9c99260a608.png">

[Source](https://www.statista.com/statistics/565270/apple-devices-ios-version-share-worldwide/)

**Comment by birkskyum** [comment]
I looked at this the other day too, and one conclusion that seems obvious is that we can drop support for iOS 9 and 10, which we still have explicit tests for.

Is 11, 12 and 13 really completely dead too? And are there any advantages for us (performance, access to new api's etc.) to dropping support for either? Even Metal was introduced in iOS 8, so we might have what we need. Whatsapp likely had more to gain from dropping iOS 11, with all the notification functionality of iOS 12

**Reply by ntadej** [reply]
Apple AFAIK supports 12 and later.

**Reply by birkskyum** [reply]
I see. Then iOS 11 would make sense to leave behind as well.

The latest iOS 11 release was in 2018, where the latest iOS 12 release was just 4 months ago, so they are clearly being treated differently

**Reply by lseelenbinder** [reply]
I think it's reasonable to drop support up to iOS 12, at least. We should probably ask our users with large user bases what they want to target to guide it as well.

**Reply by ntadej** [reply]
The question is also what changed between releases. If we do not have any conditions for iOS 11 for example, we can just keep it in, but maybe not test it.

**Reply by louwers** [reply]
Right. Support 'iOS 12 and above ' would mean it might still may work on lower iOS versions. But problems with those versions will be closed as WONTFIX, we won't run tests for them, and we might use iOS 12 APIs in any upcoming release.

**Reply by birkskyum** [reply]
Apparently "Any device that supports iOS 11 can be upgraded to iOS 12" (from ios 12 wiki page), so with that in mind I'm in favor of supporting 12+.

**Reply by ntadej** [reply]
> Right. Support 'iOS 12 and above ' would mean it might still may work on lower iOS versions. But problems with those versions will be closed as WONTFIX, we won't run tests for them, and we might use iOS 12 APIs in any upcoming release.

Not really, one should define minimum version when you build. Otherwise you can not conditionally use newer API for new versions. It also makes it clear for clients.

**Reply by louwers** [reply]
OK I suggest that I keep this discussion open for at least a week to give people a chance to respond. Otherwise we'll go with iOS 12+.

**Reply by mwilsnd** [reply]
At Meta the lowest we currently still support in select apps is iOS 12.

**Comment by roblabs** [comment]
My criteria for dropping are iOS security & Xcode tooling.  I recommend dropping iOS 11 and the associated MapLibre tests.  As well, please review the simulators used in the test bash shell scripts.

A case for dropping iOS 11:

1.  A developer would need an older version of Xcode *and* an older version of macOS.  For example, if you try to use Xcode 13.4.1 with macOS Ventura (latest), you may receive this blocking issue:  _"The version of Xcode installed on this Mac is not compatible with macOS Ventura. You have “Xcode.app” 13.4.1. Download version 14.1 for free from the App Store."_
2. Xcode 14.2 only supports simulators between iOS 12-16.  See image below.  Not that we should make a decision based on Xcode availability, but maybe Xcode is trying to tell us something?!?
3. Apple data: 4% of all devices use earlier than iOS 15.
4. If a team absolutely requires MapLibre with iOS 11, then they would rely on older releases from [maplibre/maplibre-gl-native-distribution/releases](https://github.com/maplibre/maplibre-gl-native-distribution/releases).  If they are relying on older devices with an old iOS version, then they can certainly deal with MapLibre releases that were made in the 2021-2023 era.
5. Security.  Apple made a security update for iOS 12 in the past year, but not for iOS 11.

Src:  [Apple security updates](https://support.apple.com/en-us/HT201222)
Name | Available for | Release date
| :------ | :------:  | ---: 
[iOS 12.5.6](https://support.apple.com/kb/HT213428) | iPhone 5s, iPhone 6, iPhone 6 Plus, iPad Air, iPad mini 2, iPad mini 3, and iPod touch (6th generation) | 31 Aug 2022

---

*Screen shot from Xcode 14.2 showing a range of simulators available.*
<img width=20% src="https://user-images.githubusercontent.com/118112/212089187-40a9b455-a4d1-4d4d-9168-d8be73126c0e.jpeg">

**Comment by ianthetechie** [comment]
My 2 cents is that supporting anything more than ~3 major versions can be a drag with relatively small benefit from the app developer's perspective (exceptions apply if you're something like Facebook and need to have like 95% device coverage because your target market is everyone). We should *target* the latest SDKs to avoid having to download an out of date Xcode, but can support back several versions.

If a dev needs to support an older device and is using the MapLibre SDK in their app, they can still use an older version regardless.

**Comment by louwers** [comment]
Thanks everyone for contributing to the discussion!

I created a new issue to drop support for iOS < 12. #776
